### DISCUSSION TITLE
LESS only Fomantic package is using jquery?

### DISCUSSION THREAD

**Discussion by Ray-B** [fomantic/Fomantic-UI#1717] [General]
# Bug Report

Hi, we are using the Fomantic-UI-LESS package to pull down enhancements into Semantic-UI React, and I noticed that jquery is being pulled in as a dependency.

```
    "fomantic-ui-less": {
      "version": "2.8.7",
      "resolved": "https://registry.npmjs.org/fomantic-ui-less/-/fomantic-ui-less-2.8.7.tgz",
      "integrity": "sha512-qCZmm/eG3cvJOfJmyFYkM+iP4BlBX+hflwQ3mxNJg7QnClYao5N1ff6XrOQVWV8at10etYmSoJ7Q70YZRBKPkQ==",
      "dev": true,
      "requires": {
        "jquery": "^3.4.0"
      }
    },
```

I realize that the _complete_ fomantic-ui library obviously needs jquery for now, but I was surprised to discover jquery when all we want is the LESS. 

Is this required? Workaround? Or something we can contribute a PR for?

Thanks!

## Steps to reproduce
1.  Add fomantic-ui-less
2. node_modules will now contain jquery
3.

## Expected result
I would have expected if all I am using is the eventually compiled/transpiled LESS, that jquery would not be needed

## Actual result
JQuery is being added, and adding substantial cruft to our release package.

## Testcase
N/A

## Screenshot (if possible)
N/A

## Version
2.8.7

**Comment by lubber-de** [comment]
The LESS Repo also includes the related js modules, to have a complete working fomantic package. That's why.
Of course don't need those js modules (and jquery) when using SUI React.

@hammy2899 What do you think? If we remove the jquery dependency from the package.json, shouldn't we remove the js modules inside the LESS package as well then? We have a similar situation in the CSS repo. 🤔

**Comment by Ray-B** [comment]
I'm not opposed to issuing a PR for this, FWIW

**Comment by TheJltres** [comment]
SUI React also loads jQuery using `semantic-css` (https://github.com/Semantic-Org/Semantic-UI-React/blob/master/package.json#L170).

This change, would leave all (CSS ?) LESS users without JS modules, that is a big change.
This can be seen also from JS modules perspective, there is no way of installing fomantic with only JS.

How would you use LESS and JS? You can use CSS and JS from Fomantic repo, but there would not be any fomantic less repo with JS.
What I think would be correct, is move all CSS/LESS JS modules, to a new repo 😪... Like this you could have CSS/LESS with or without JS

**Comment by lubber-de** [comment]
@Ray-B How are you packaging your final builds? Having another folder for jquery inside node_module isn't really an issue at install time, but what counts is the final package you use for your project. So maybe you can configure your packaging tool to exclude the "node_modules/jquery" folder?

**Comment by TheJltres** [comment]
You can also fork the repository and delete that dependecy or all JS modules.

**Comment by Ray-B** [comment]
A few comments:

@lubber-de Packaging is handled by AWS Codebuild, with release packages being pushed to S3. Stripping the unused package out is temporary option; There is still a cost to pulling down this package every time a branch or release build is done, even if it doesn't end up in the release. Theres also the fact that we're then writing some very "special" code just to strip out a package, which is a bit weird.

I think your idea of separating the LESS/CSS/SASS from the JS is a good one. I've seen this pattern followed on several UI libraries, and I think it generally makes for a less coupled and more flexible library. Getting this sorted seems like a worthwhile expenditure of time.

@TheJItres Forking is an option, definitely. Although if we fix it once in the main, then nobody has to bother with the maintenance of their particular fork. 👍 

Thanks for the feedback on this. Let me know how I can be of help to push this forward.

**Comment by lubber-de** [comment]
Just found #1039 which is somehow requesting the opposite (having the LESS repo include the whole semantic.js which would be something against moving out the js modules... i am not argueing, i just want to have all related issues mentioned/considered for a overall solution

**Reply by Ray-B** [reply]
Definitely. I think there is a good reason for needing both. 

From a development perspective, reduced coupling makes the most sense, so having them logically separate would be good. Would allow for the development of the semantic portion separate from the implementation, e.g. vue, react, plain old javascript, etc. 

One possibility might be to provide a release package with it bundled for users who require both. An npm package to facilitate this would probably do the trick:

e.g. `npm install fomantic-ui-less`  `npm install fomantic-ui-js` `npm install fomantic-ui-vue` etc.
